src/Workspaces/Remote/Core/Telemetry/ApiUsageIncrementalAnalyzerProvider.cs

```diff
@@ -0,0 +1,211 @@
+                        }
+
+                        // this only gather reference and method call symbols but not type being used.
+                        // if we want all types from metadata used, we need to add more cases 
+                        // which will make things more expansive.
+                        CollectApisUsed(operation, metadataSymbolUsed);
+                    }
+                }
+
+                var groupByAssembly = metadataSymbolUsed.GroupBy(s => s.ContainingAssembly);
+                var apiPerAssembly = groupByAssembly.Select(g => new
+                {
+                    // mark all string as PII (customer data)
+                    AssemblyName = new TelemetryPiiProperty(g.Key.Identity.Name),
+                    AssemblyVersion = g.Key.Identity.Version.ToString(),
+                    Symbols = g.Select(s => s.GetDocumentationCommentId()).Where(id => id != null).Select(id => new TelemetryPiiProperty(id))
+                });
+
+                lock (_reported)
+                {
+                    if (_reported.Add(project.Id))
+                    {
+                        // use telemetry API directly rather than Logger abstraction for PII data
+                        var telemetryEvent = new TelemetryEvent(EventName);
+                        telemetryEvent.Properties[PropertyName] = new TelemetryComplexProperty(apiPerAssembly);
+
+                        try
+                        {
+                            RoslynServices.SessionOpt?.PostEvent(telemetryEvent);
+                        }
+                        catch
+                        {
+                            // don't crash OOP because we failed to send telemetry
+                        }
+                    }
+                }
+
+                return;
+
+                // local functions
+                static void CollectApisUsed(IOperation operation, HashSet<ISymbol> metadataSymbolUsed)
+                {
+                    switch (operation)
+                    {
+                        case IMemberReferenceOperation memberOperation:
+                            AddIfMetadataSymbol(metadataSymbolUsed, memberOperation.Member);
+                            break;
+                        case IInvocationOperation invocationOperation:
+                            AddIfMetadataSymbol(metadataSymbolUsed, invocationOperation.TargetMethod);
+                            break;
+                        case IObjectCreationOperation objectCreation:
+                            AddIfMetadataSymbol(metadataSymbolUsed, objectCreation.Constructor);
+                            break;
+                    }
+                }
+
+                static void AddIfMetadataSymbol(HashSet<ISymbol> metadataSymbolUsed, ISymbol symbol)
+                {
+                    // get symbol as it is defined in metadata
+                    symbol = symbol.OriginalDefinition;
+
+                    if (metadataSymbolUsed.Contains(symbol))
```

Comment:
note: you might want an equality comparer that unifies over things like Nullable so you don't rereport the same member.

Reply:
yep. good catch. didn't know that new nullability feature now give different symbol based on nullability annotation.